src/rawArray/index.ts

```diff
@@ -0,0 +1,94 @@
+import { DtypeString } from '../types';
+import { ArraySelection } from '../core/types';
+import { slice } from '../core/slice';
+import { ValueError } from '../errors';
+import { normalizeShape, IS_NODE, getStrides } from '../util';
+import { TypedArray, DTYPE_TYPEDARRAY_MAPPING, getTypedArrayDtypeString, TypedArrayConstructor } from '../nestedArray/types';
+import { setRawArrayDirect, setRawArrayToScalar } from './ops';
+
+export class RawArray {
+    dtype: DtypeString;
+    shape: number[];
+    strides: number[];
+    data: TypedArray;
+
+    constructor(data: TypedArray, shape?: number | number[], dtype?: DtypeString, strides?: number[])
+    constructor(data: Buffer | ArrayBuffer | null, shape?: number | number[], dtype?: DtypeString, strides?: number[])
+    constructor(data: Buffer | ArrayBuffer | TypedArray | null, shape?: number | number[], dtype?: DtypeString, strides?: number[]) {
```

Comment:
Note for myself: one day this and the nestedarray constructor should be cleaned up

Reply:
I'll add an issue!